### ISSUE TITLE
Experiment with enabling --include-styles

### ISSUE THREAD

**Issue by bfirsh** [arxiv-vanity/engrafo#270]
LaTeXML doesn't include styles that don't have a "binding" that descibes how LaTeXML should render it. LaTeXML is getting good enough at TeX parsing now though that it might be able to parse style files itself.

It has an optional flag for this: `--include-styles`. We should render a set of recent papers with this option on and compare the error rate to renders without the option.

**Comment by dginev** [comment]
Experimenting with raw package interpretation is something Bruce and I have been meaning to do for a while. We got around to trying it out recently, for the arXiv sandbox we are playing with (the Feb 2018 month).

For now the results aren't exciting (the `--includestyles` run tends to fail too often), but we'll use the data to improve the raw machinery.

1. regular run: https://corpora.mathweb.org/corpus/arxiv_1802/tex_to_html
2. run with raw packages: https://corpora.mathweb.org/corpus/arxiv_1802/raw_tex_to_html

Also a warning for future readers - these links are fragile, we use the sandbox for current development and it may disappear at some point down the line.

**Comment by dginev** [comment]
If we obtain a sandbox of arxiv papers that are of high interest to arxiv-vanity, I can set that up as a standalone corpus and do the comparison - but i expect it will have a similar flavor.

**Comment by dginev** [comment]
Interesting experiment. and it sounds about right.

The lack of styling reminds me of the reason binding macro definitions are allowed to be `locked` by latexml. Exactly to protect redefinitions based on low-level tex, as one would expect in raw class and style files.

As to picking up the stuff that actually matters and ignoring the rest - completely on board. It's tricky because there is a big long-tail of constructs to add "ignored" support for. We still are missing some small corners of the `latex.ltx` core definitions for instance, but I am tempted to grab your example and see if I can land a couple raw interpretations upgrades to make it work nicely.

**Comment by bfirsh** [comment]
Funnily, the styling issue doesn't affect Engrafo because it blindly applies its stylesheet to everything. I was testing with plain LaTeXML...

So, from the look of the cortex results:

- There are a few more renders without problems when `--includestyles is enabled, presumably from a few packages that are missing bindings but latexml parses and runs correctly.
- A lot more fatal renders, presumably from missing bindings that latexml doesn't parse correctly.

That means we'd be currently worse off enabling `--includestyles`, right?

I wonder if something like this would work:

1. Attempt to render paper with `--includestyles`
2. If fatal error, render without `--includestyles`

That way, we get a few more papers with no problems, but if there is a fatal problem parsing, it just reverts back to not trying to parse the package and showing an error. 🤔

**Comment by dginev** [comment]
Right, the `--includestyles` runs are both less predictable and more prone to errors at the moment, so you may want to be extra conservative in using the switch. The "fallback run" approach is interesting, may be an interesting performance vs coverage trade-off. 

On the cortex side, my intention is to keep running subsets of arXiv with parallel on/off switch states and compare the results as latexml develops. At some point the raw interpretation will be reliable enough to overtake the current default. We've also talked internally that when the time comes it would also make sense to have `--includestyles` on by default, but we're not there just yet.

**Comment by bfirsh** [comment]
1711.10462 has a local `macros.sty` file that defines all the macros used in the document. All simple stuff that LaTeXML can handle, but it isn't loaded because we don't use `--includestyles`.

I wonder if there is something simple we can do to make this work, without going all-in on `--includestyles`...

**Comment by dginev** [comment]
Well, not really... That .sty file may be simple, but a different paper may have a local style file that completely destroys the conversion with a Fatal error, arxiv is quite wild.

And it's not even easy to say "load this raw, if it causes no issues", as latex packages are closer to instructions to a stack machine than namespaced modules with definitions. If you law a stay and latexml trips on it, there is no simple way to "unload" it, it may have corrupted the state in subtle and unpleasant ways. But maybe I'm missing some smart trade-off somewhere.
